### PR TITLE
SWFFilters.py: allow X.Y.Z-Codename-GitRevision

### PR DESCRIPTION
**Pull request by tobsan**
The codename (such as alpha, beta, or possibly pyro) was not allowed
before, causing the spell check to crash on the PELUX Baseline SWF.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

### CODE REVIEW

File: docs/python_modules/enchantfilters/SWFFilters.py [hunk 1]
```diff
@@ -46,8 +47,18 @@ class VersionStringFilter(Filter):
...
-        (version,sep,revparse) = word.partition('-')
+        (version,sep,rest) = word.partition('-')
         if version == word:
             return False # probably not a version string then
         else:
-            return all_match(version, self.versionchars) and all_match(revparse, self.allowedchars)
+            # The version string has to look right
+            if not all_match(version, self.versionchars):
+                return False
+            else:
```
Comment (demonicblue) [inline docs/python_modules/enchantfilters/SWFFilters.py]: Try removing this for easier reading

File: docs/python_modules/enchantfilters/SWFFilters.py [hunk 2]
```diff
@@ -46,8 +47,18 @@ class VersionStringFilter(Filter):
     versionchars = string.digits + "."
 
     def _skip(self, word):
-        (version,sep,revparse) = word.partition('-')
+        (version,sep,rest) = word.partition('-')
         if version == word:
             return False # probably not a version string then
         else:
```
Comment (demonicblue) [inline docs/python_modules/enchantfilters/SWFFilters.py]: Try removing this for easier reading

**Review by demonicblue — COMMENTED** [review]
I find the nesting a bit hard to read. Could it be a good idea to remove the else-statements or would it diverge from the pythonic way?

File: docs/python_modules/enchantfilters/SWFFilters.py (hunk 2 above)
Comment (tobsan) [reply docs/python_modules/enchantfilters/SWFFilters.py]: I kept this one actually but I removed the others.

**Review by demonicblue — APPROVED** [review]

**Pull request merged**